src/API/AbstractAPIClient.php

```diff
@@ -30,24 +29,29 @@ public function __construct(IntegrationInterface $integration)
 
     /**
      * @param Request $request
+     *
      * @return array|mixed
      */
     public function callAPI(Request $request)
     {
         try {
+            if ($this->getAPIClientName() == 'PLUGIN API') {
+                $this->logger->error('Plugin API can not be called');
+            }
```

Comment:
This is implementation specific logic which we want to keep out of the abstract class.  Instead of implementing this logic here lets extend callAPI in CF\API\Plugin:
```
public function callAPI(Request $request) {
   return $this->createAPIError("The url: " . $request->getUrl() . " is not a valid path.");
}
```